projects/pandas/fuzz_pickle_parser.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This fuzzer script targets the pandas pickle parser, attempting to deserialize a wide range of fuzzed byte streams."""
+
+import sys
+import atheris
+import pandas as pd
+import io
+
+def TestPickleInput(data):
+    fdp = atheris.FuzzedDataProvider(data)
+
+    try:
+        length = fdp.ConsumeIntInRange(1, len(data))
+        fuzzed_data = fdp.ConsumeBytes(length)
+        fuzzed_io = io.BytesIO(fuzzed_data)
+        pd.read_pickle(fuzzed_io)
+    except Exception as e:
```

Comment:
What's the goal in terms of bugs to find considering we catch all exceptions (same applies to the other fuzzers)?

Also, is this effectively fuzzing pickle? If so, Pickle is inherently insecure: https://docs.python.org/3/library/pickle.html#:~:text=The%20pickle%20module%20is%20not,could%20have%20been%20tampered%20with. so am not sure what we're going for in that case?

Reply:
Hi @DavidKorczynski Exceptions were updated and adjusted according to functions used in the fuzzers. 
According pickle fuzzer, I was working with pickle parser and I have got Out-Of-Memory crash from the fuzzing engine, which was classical example of`Billion laughs attack`. I forgot to remove it from the actual PR.